dev/release/05-binary-upload.sh

```diff
@@ -135,3 +136,35 @@ docker_run \
     VERBOSE=${VERBOSE:-no} \
     VERSION=${version} \
     YUM_TARGETS=$(IFS=,; echo "${yum_targets[*]}")
+
+# Upload the MATLAB MLTBX Release Candidate to the GitHub Releases
+# area of the Apache Arrow GitHub project.
+mltbx_file="${ARROW_ARTIFACTS_DIR}/matlab-dist/matlab-arrow-${version_with_rc}.mltbx"
+mltbx_signature_gpg_ascii_armor=${mltbx_file}.asc
+mltbx_checksum_sha512=${mltbx_file}.sha512
+
+# Sign the MLTBX file and create a detached (--deatch-sign) ASCII armor (--armor) GPG signature file.
+gpg --detach-sign --local-user "${GPG_KEY_ID}" --armor ${mltbx_file}
+
+# Compute the SHA512 checksum of the MLTBX file.
+shasum --algorithm 512 ${mltbx_file} > ${mltbx_checksum_sha512}
+
+# Upload artifacts to the Apache Arrow
+# GitHub Releases area and mark as a "Prerelease".
+if [ ${UPLOAD_MATLAB} -gt 0 ]; then
+  release_tag=apache-arrow-${version_with_rc}
+  target_branch=release-${version_with_rc}
+  release_notes="Release Candidate: ${version} RC${rc}"
+  title="Apache Arrow ${version} RC${rc}"
+  repository="https://github.com/apache/arrow"
+  gh release create \
```

Comment:
this means we requires the gh CLI installed locally, right?
Can we add a note on the requirements for the release tasks that we require gh CLI installed?
https://github.com/apache/arrow/blob/main/docs/source/developers/release.rst

Reply:
Hi @raulcd,

This change is out of date - sorry about that. Based on @kou's feedback, we going to use a GitHub Actions Workflow - triggered by pushing the tag `apache-arrow-X.Y.Z-rc{N}` - to create release instead of using `gh`. I just haven't updated `05-binary-upload.sh` yet, but will do so soon!

Reply:
Actually, I just realized you are right. We will need to add a note about requiring the GItHub CLI to be installed locally. While we are no longer using `gh` to create the release in `04-binary-upload.sh`, we will need to use `gh` once the GitHub release is created to upload the signed MLTBX files.  

I'll make sure to add a note about this requirement to [documentation](https://github.com/apache/arrow/blob/main/docs/source/developers/release.rst).

Sorry about the confusion!